Support sp_rename for additional OBJECT type (TRIGGER, TABLE_TYPE)

raydhkim opened the pull request:
Previously, TRIGGER and TABLE_TYPE were not supported for sp_rename procedure. Now, these two subtypes are included in OBJECT objtype of sp_rename, allowing users to rename TRIGGER and TABLE_TYPE objects using sp_rename.

Task: BABEL-3965

### Description

[Describe what this change achieves - Guidelines below (please delete the guidelines after writing the PR description)]

> 1. *What* is the change? This is best described in terms of “Currently, Babelfish does X. With this change it now does Y.” Think of “What *did* it *used* to do?” and “What *does* it do *now*?”
2. *Why* was the change made? What drove our desire to put effort into the change?
3. *How* was the code changed should only appear for large commits. This can serve as a rough roadmap to what’s contained in the commit. It should be very high level; if it’s directly referencing code it’s probably too detailed. It’s also critical that this section of a commit message does not try to replace proper code documentation (ie, block comments or README files). Generally, this section should only appear if the commit itself is large enough that it’s helpful to provide a roadmap to someone looking at the commit.
4. The last descriptive piece is the “title” for the commit: the very first line of the commit message, which should typically be less than 80 characters. A good title is *critical*, because it’s the only thing that shows up in places like the Github commit listing. No one’s got time to read through full commit messages when trying to find a single commit out of dozens.


### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**


* **Boundary conditions -**


* **Arbitrary inputs -**


* **Negative test cases -**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

raydhkim committed:
Support sp_rename for additional OBJECT type (TRIGGER, TABLE_TYPE)

Previously, TRIGGER and TABLE_TYPE were not supported for sp_rename procedure. Now, these two subtypes are included in OBJECT objtype of sp_rename, allowing users to rename TRIGGER and TABLE_TYPE objects using sp_rename.

Task: BABEL-3965

Signed-off-by: Ray Kim <raydhkim@amazon.com>

raydhkim committed:
Merge remote-tracking branch 'upstream/BABEL_3_X_DEV' into babel-3965-2

Signed-off-by: Ray Kim <raydhkim@amazon.com>

lejaokri commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -2576,7 +2580,7 @@ rename_procfunc_update_bbf_catalog(RenameStmt *stmt)
 								  bbf_func_ext_rel->rd_att, &is_null);
 	funcsign = pstrdup(TextDatumGetCString(funcsign_datum));
-	new_funcsign = strcat(pstrdup(stmt->newname), strrchr(funcsign, '('));
+	snprintf(new_funcsign, sizeof(new_funcsign), "%s%s", stmt->newname, strrchr(funcsign, '('));
```
please unit test where length of new_funcsign could go beyond NAMEDATALEN

lejaokri commented on test/JDBC/jdbc_schedule:
```diff
@@ -122,8 +122,3 @@ ignore#!#openquery_upgrd-vu-cleanup
 ignore#!#Test-Identity-before-14_7-or-15_2-vu-verify
 ignore#!#Test-Identity-before-14_7-or-15_2-vu-cleanup
-
-# These tests are meant for upgrade scenario prior to (potential) 14_8 or 15_3 release
-ignore#!#BABEL-4078-before-14_8-or-15_3-vu-prepare
-ignore#!#BABEL-4078-before-14_8-or-15_3-vu-verify
-ignore#!#BABEL-4078-before-14_8-or-15_3-vu-cleanup
```
this change shouldnt be here

lejaokri commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -2522,12 +2526,12 @@ rename_procfunc_update_bbf_catalog(RenameStmt *stmt)
+	char		new_funcsign[NAMEDATALEN] = {0};
 
 	/* build the tuple to insert */
 	MemSet(new_record_func_ext, 0, sizeof(new_record_func_ext));
```
I dont think we need these MemSets if we initialize the variables as 0

```
Datum	new_record_func_ext[BBF_FUNCTION_EXT_NUM_COLS] = {0};
bool		new_record_nulls_func_ext[BBF_FUNCTION_EXT_NUM_COLS] = {false};
bool		new_record_repl_func_ext[BBF_FUNCTION_EXT_NUM_COLS] = {false};
```

lejaokri commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -2576,7 +2580,7 @@ rename_procfunc_update_bbf_catalog(RenameStmt *stmt)
+	snprintf(new_funcsign, sizeof(new_funcsign), "%s%s", stmt->newname, strrchr(funcsign, '('));
 
 	new_record_func_ext[Anum_bbf_function_ext_funcname - 1] = CStringGetDatum(stmt->newname);
 	new_record_func_ext[Anum_bbf_function_ext_orig_name - 1] = CStringGetTextDatum(orig_proc_funcname);
```
Can `orig_proc_funcname` be NULL? If not, please add an `Assert(orig_proc_funcname)` here because we're using some global variable that this function never set

xhfanhe replied:
If orig_proc_funcname is NULL, I think we don't need to update this column. Example: I remember the function created for trigger doesn't have a orig_funcname

xhfanhe commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -2520,12 +2524,12 @@ rename_procfunc_update_bbf_catalog(RenameStmt *stmt)
 	Node	   *schema;
 	char	   *schemaname;
 	ObjectWithArgs *objwargs = (ObjectWithArgs *) stmt->object;
+	char		new_funcsign[NAMEDATALEN] = {0};
```
funcsign column is TEXT, shouldn't be restricted by NAMEDATALEN

raydhkim committed:
ssh-ing into github prep

raydhkim committed:
Revert "ssh-ing into github prep"

This reverts commit a338eeb53cb1165aec62b5113918b7d6759a5ccb.

raydhkim committed:
Support sp_rename for additional OBJECT type (TRIGGER, TABLE_TYPE)

Previously, TRIGGER and TABLE_TYPE were not supported for sp_rename procedure. Now, these two subtypes are included in OBJECT objtype of sp_rename, allowing users to rename TRIGGER and TABLE_TYPE objects using sp_rename.

Task: BABEL-3965

Signed-off-by: Ray Kim <raydhkim@amazon.com>

xhfanhe commented on test/JDBC/upgrade/14_8/schedule:
```diff
@@ -399,4 +399,4 @@ datetime2fromparts
 orderby-before-15_3
 BABEL-3215
-BABEL-4078-before-14_8-or-15_3
+BABEL-4078
```
This change should already be merged, would you please try a branch merge?

xhfanhe commented on test/JDBC/jdbc_schedule:
```diff
@@ -8,7 +8,10 @@
+# all
+Test-sp_rename-vu-prepare
+Test-sp_rename-vu-verify
+Test-sp_rename-vu-cleanup
```
Please remove these unwanted changes

xhfanhe commented on contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.1.0--3.2.0.sql:
```diff
@@ -1745,6 +1745,7 @@ CREATE OR REPLACE PROCEDURE sys.sp_rename(
 LANGUAGE 'pltsql'
 AS $$
 BEGIN
+	--trying ssh
```
Unwanted change

raydhkim committed:
Merge remote-tracking branch 'upstream/BABEL_3_X_DEV' into babel-3965-2

Signed-off-by: Ray Kim <raydhkim@amazon.com>
